Tasks/Common/ServiceFabricHelpers/Connect-ServiceFabricClusterFromServiceEndpoint.ps1

```diff
@@ -185,8 +185,12 @@ function Connect-ServiceFabricClusterFromServiceEndpoint
         # Configure cluster connection pre-reqs
         if ($ConnectedServiceEndpoint.Auth.Scheme -ne "None")
         {
-            # Add server cert thumbprint(s) (common to both auth-types)
-            if ($ConnectedServiceEndpoint.Auth.Parameters.ServerCertThumbprint)
+            # Add server cert thumbprint(s)/commonname(s) (common to both auth-types)
+            if ($ConnectedServiceEndpoint.Auth.Parameters.CertLookUp -and $ConnectedServiceEndpoint.Auth.Parameters.CertLookUp -eq "CommonName" -and $ConnectedServiceEndpoint.Auth.Parameters.ServerCertCommonName) 
```

Comment:
Will checking just `if ($ConnectedServiceEndpoint.Auth.Parameters.ServerCertCommonName)` be sufficient? Can it happen that LookUp is set to "Thumbprint" but still CertCommonName is filled up. I am not sure but this might happen if user selected "Common name" in look up field, filled up a thumbprint and then switched to "thumbprint" in look up field and filled thumbprint as well. If in this case both $ConnectedServiceEndpoint.Auth.Parameters.ServerCertCommonName and $ConnectedServiceEndpoint.Auth.Parameters.ServerCertThmbprint are non-null it can be problematic in some of the checks you have added in PR

Reply:
Yes, checking $ConnectedServiceEndpoint.Auth.Parameters.ServerCertCommonName should be sufficient. That was the idea in my original spec but changed it to include certlookup check as well based on feedback.